cordformation/src/main/kotlin/net/corda/plugins/Signing.kt

```diff
@@ -0,0 +1,65 @@
+package net.corda.plugins
+
+import org.gradle.api.tasks.Input
+import java.nio.file.Path
+
+//contains union of properties for ANT tasks "genkey" and "signjar" as most of them are shared by both tasks
+class Signing {
+
+    var defaultKeyStoreFileName = "jarSignKeystore.jks"
+    var defaultStoretype = "jks"
+    var defaultStorepass = "secret1!"
+    var defaultAlias = "cordapp-signer"
+    var defaultDname = "OU=Dummy Cordapp Distributor, O=Corda, L=London, C=GB"
+
+    @get:Input
+    var all: Boolean = false
+        private set
+    @get:Input
+    var enabled: Boolean = true
+        private set
+    @get:Input
+    var generateKeystore: Boolean = true
+        private set
+
+    private var opts = mutableMapOf("storetype" to defaultStoretype,
```

Comment:
Judging from the "allowedOpts" list this should just be another data class so we can keep the whitelist statically typed.